Example/ChatWallet/PresentationLayer/Wallet/Wallet/WalletView.swift

```diff
@@ -0,0 +1,65 @@
+import SwiftUI
+
+struct WalletView: View {
+    let interactor: WalletInteractorProtocol
```

Comment:
In classic VIPER patter Interactor is to interact with services.  
I agree, now interactor is looks like redundant but this is only because the app on the initial stage and we does not have a lot services. In future we will have more storage logic, push notification services, network services etc.

If we injecting interactor in the View layer, that means what we are mixing service logic with view specific logic (filtering, updating view, reloading etc). 

If you fill what interactor is redundant I think is better to just remove it (if it not needed for this module), then transfer view events into it instead of presenter.

Reply:
If you tap, f.i. on some filter button, you pass you selection from **Presenter** as a parameter to the **Interactor**. **Interactor** does its job and updates view via **Presenter**. Is there some mixing logic with **View**?

My point was mostly about the fact that app's business logic is spread between **Presenter** and **Interactor** and it doesn't depend on how much services, etc. Since **Presenter** in **SwiftUI** is a bit different, I wouldn't try to fit some "classic VIPER" and stuff, just stick to some clean solution.

But yeah removing the **Interactor** I believe is only a temporary solution. As you've mentioned if we get a bigger app, **Presenter** will grow up and things get messy.